azure-quantum/azure/quantum/workspace.py

```diff
@@ -374,8 +374,18 @@ def login(self, refresh:bool=False) -> Authentication:
             the user credentials to authenticate with Azure Quantum
         """
 
-        if refresh or (self.credentials is None):
-            msal_wrapper = MsalWrapper(self.subscription_id, refresh=refresh)
-            self.credentials = msal_wrapper.acquire_auth_token()
+        if refresh:
+            self.credentials = None
 
-        return BasicTokenAuthentication(self.credentials)
+        if self.credentials is None:
+            msal_wrapper = MsalWrapper(subscription_id=self.subscription_id, refresh=refresh)
+
+            auth_token = msal_wrapper.acquire_auth_token()
+            return BasicTokenAuthentication(token=auth_token)
+
+
+        if not isinstance(self.credentials, Authentication):

```

Comment:
Is this for the case where credentials is a ServicePrincipalCredentials? If so, could you please add a comment saying that?

Reply:
Done

Reply:
Curious, what is the scenario where do need to wrap it with a BasicTokenAuthentication? Is it just for the test scenario? If so, why could we not have the caller/test do that?

Reply:
The test scenario does not use that path to wrap in a BasicTokenAuthentication.
I left it there just in case the user provides some type of credential that would require the wrapping.
We will replace the auth flow in March as we move to use Azure Identity and track2 generated code, so I just wanted to make sure we were not breaking some working scenario with this fix.